lms/djangoapps/django_comment_client/tests/test_utils.py

```diff
@@ -612,11 +626,50 @@ def test_ids_mixed(self):
         self.create_discussion("Chapter 2", "Discussion")
         self.create_discussion("Chapter 2 / Section 1 / Subsection 1", "Discussion")
         self.assertItemsEqual(
-            utils.get_discussion_categories_ids(self.course),
+            utils.get_discussion_categories_ids(self.course, self.user),
             ["Topic_A", "Topic_B", "Topic_C", "discussion1", "discussion2", "discussion3"]
         )
 
 
+class ContentGroupCategoryMapTestCase(CategoryMapTestMixin, ContentGroupTestCase):
+    """
+    Tests `get_discussion_category_map` on discussion modules which are
+    only visible to some content groups.
+    """
+    def test_alpha_and_staff_can_see(self):
+        for user_with_access in [self.alpha_user, self.staff_user]:
+            self.assert_category_map_equals(
```

Comment:
Building upon my above post, I believe that the assumption is that the user is already verified to be in the course by logic in the forums views. Of course that assumption is based on the assumption that this code is only called from forum views. So yeah, it does seem a little strange.